ssl/t1_lib.c

```diff
@@ -3317,7 +3317,7 @@ int tls_choose_sigalg(SSL *s, int fatalerrs)
                 if (i == sent_sigslen) {
                     if (!fatalerrs)
                         return 1;
-                    SSLfatal(s, SSL_AD_ILLEGAL_PARAMETER,
+                    SSLfatal(s, SSL_AD_HANDSHAKE_FAILURE,
```

Comment:
This one seems like a good change; we are doing a negotiation, just using the protocol default for the peer's offer, and found no overlap.  That's a handshake failure, not an illegal parameter.